locals.tf

```diff
@@ -5,86 +5,428 @@ locals {
     lookup(module.schema_label.descriptors, var.descriptor_name, module.schema_label.id), "/${module.schema_label.delimiter}${module.schema_label.delimiter}+/", module.schema_label.delimiter
   ), module.schema_label.delimiter) : null
 
-  create_default_roles = module.this.enabled && var.create_default_roles
+  schema   = coalesce(one(snowflake_schema.this[*].name), var.name)
+  database = var.database
 
-  #This needs to be the same as an object in roles variable
+  # This needs to be the same as an object in roles variable
   role_template = {
-    enabled                        = true
-    comment                        = null
-    role_ownership_grant           = "SYSADMIN"
-    granted_roles                  = []
-    granted_to_roles               = []
-    granted_to_users               = []
-    add_grants_to_existing_objects = var.add_grants_to_existing_objects
-    schema_grants                  = []
-    table_grants                   = []
-    dynamic_table_grants           = []
-    external_table_grants          = []
-    view_grants                    = []
-    materialized_view_grants       = []
-    file_format_grants             = []
-    function_grants                = []
-    stage_grants                   = []
-    task_grants                    = []
-    procedure_grants               = []
-    sequence_grants                = []
-    stream_grants                  = []
+    enabled                   = true
+    role_ownership_grant      = "SYSADMIN"
+    granted_to_roles          = []
+    granted_database_roles    = []
+    granted_to_database_roles = []
+    schema_grants             = []
+    database_grants           = []
+    schema_objects_grants = {
+      "TABLE"             = []
+      "DYNAMIC TABLE"     = []
+      "EXTERNAL TABLE"    = []
+      "VIEW"              = []
+      "MATERIALIZED VIEW" = []
+      "FILE FORMAT"       = []
+      "FUNCTION"          = []
+      "STAGE"             = []
+      "TASK"              = []
+      "PROCEDURE"         = []
+      "SEQUENCE"          = []
+      "STREAM"            = []
+    }
   }
 
-  default_roles_definition = {
-    readonly = {
-      schema_grants            = ["USAGE"]
-      table_grants             = ["SELECT"]
-      dynamic_table_grants     = ["SELECT"]
-      external_table_grants    = ["SELECT", "REFERENCES"]
-      view_grants              = ["SELECT", "REFERENCES"]
-      materialized_view_grants = ["SELECT", "REFERENCES"]
-      file_format_grants       = ["USAGE"]
-      function_grants          = ["USAGE"]
-      stage_grants             = ["USAGE"]
-      task_grants              = ["MONITOR"]
+  default_roles_definition = var.create_default_database_roles ? {
+    admin = {
+      schema_grants = [{
+        privileges                 = ["ALL PRIVILEGES"]
```

Comment:
We should use this flag `all_privileges = true` instead of `privileges = ["ALL PRIVILEGES"]`

Reply:
Setting this flag breaks conditional `default_role_definition` creation - I would like to stick to this formulae. 
Looking at provider files, `["ALL PRIVILEGES"]` are respected: https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/templates/resources/grant_privileges_to_database_role.md.tmpl#L41

Reply:
Since that works fine - we can stick with your solution